### DISCUSSION TITLE
How to set draw Z-ordering in a layer

### DISCUSSION THREAD

**Discussion by zsoooc** [visgl/deck.gl#9297] [Q&A]
I would like to set the Z-ordering on my polygon layer (GeoJsonLayer) . There is a "category" attribute with values: 1 and 2. Polygons with category 2 should be on the top.

How can I set Z ordering?

**Comment by ibgreen** [comment]
Make sure you have read https://deck.gl/docs/developer-guide/tips-and-tricks

It is generally not very straightforward to adjust the position of each element in a layer
- One technique is to use two polygon layers 
  - draw each layer with slightly different z positions 
  - or disabling depth test (at least for the second layer), ensuring that second layer draws over first layer.
- Alternatively, you can modify z positions very slightly for individual polygons based on the category value.
- For the GPU programmer, it might be possible to create a deck.gl extension that adjusts heights in the vertex shader.
